src/main/java/org/photonvision/SimVisionSystem.java

```diff
@@ -0,0 +1,139 @@
+     * to the subste of these targets which are visible from the given robot position.
+     * @param tgt
+     */
+    public void addSimVisionTarget(SimVisionTarget tgt){
+        tgtList.add(tgt);
+    }
+
+    /**
+     * Adjust the camera position relative to the robot.
+     * Use this if your camera is on a gimbal or turret or some other mobile platform.
+     * @param newRobotToCamera New Tranform from the robot to the camera
+     * @param newCamHeight New height of the camera off the floor
+     */
+    public void moveCamera(Transform2d newRobotToCamera, double newCamHeight){
+        this.robotToCamera  = newRobotToCamera;
+        this.cameraHeightOffGround_m  = newCamHeight;
+    }
+
+    /**
+     * Periodic update. Call this once per frame of image data you wish to process and send to NetworkTables
+     * @param robotPose current pose of the robot on the field. Will be used to calcualte which targets are actually in view, where they are at relative to the robot, and relevant PhotonVision parameters.
+     */
+    public void processFrame(Pose2d robotPose){
+
+        Pose2d cameraPos = robotPose.transformBy(robotToCamera);
+
+        LinkedList<PhotonTrackedTarget> visibleTgtList = new LinkedList<PhotonTrackedTarget>();
+
+        tgtList.forEach((tgt) -> {
+            var camToTargetTrans = new Transform2d(cameraPos, tgt.targetPos);
+
+            double dist_along_ground_m = camToTargetTrans.getTranslation().getNorm();
+            double dist_vertical_m = tgt.targetHeight_m - this.cameraHeightOffGround_m;
+            double dist_m = Math.hypot(dist_along_ground_m, dist_vertical_m);
+
+            double area_px = tgt.tgtArea_m2 / getM2PerPx(dist_along_ground_m);
+
+            double yaw_deg = PhotonUtils.wrapAngleDeg(camToTargetTrans.getRotation().getDegrees());
+            double pitch_deg = Math.atan2(dist_vertical_m, dist_along_ground_m) - this.camPitch_deg;
+
+            if(camCanSeeTarget(dist_m, yaw_deg, pitch_deg, area_px)){
+                visibleTgtList.add(new PhotonTrackedTarget(yaw_deg, pitch_deg, area_px, 0.0, camToTargetTrans));
+            }
+        });
+
+        cam.submitProcessedFrame(0.0, visibleTgtList);
+
+    }
+
+    double getM2PerPx(double dist){
+        double widthMPerPx  = 2 * dist * Math.tan(this.camHorizFOV_deg/2) / this.cameraRes_px.getFirst();
+        double heightMPerPx = 2 * dist * Math.tan(this.camVertFOV_deg/2)  / this.cameraRes_px.getSecond();
+        return widthMPerPx * heightMPerPx;
+    }
+
+
+    boolean camCanSeeTarget(double dist_m, double yaw, double pitch, double area){
+        boolean inRange = (dist_m < this.maxLEDRange_m);
+        boolean inHorizAngle = Math.abs(yaw) < (this.camHorizFOV_deg/2);
+        boolean inVertAngle = Math.abs(pitch) <  (this.camVertFOV_deg/2);
+        boolean targetBigEnough = area > this.minTargetArea;
+        return (inRange & inHorizAngle & inVertAngle & targetBigEnough);
```

Comment:
Should these be `&&`?

Reply:
Yes https://www.tutorialspoint.com/Java-Boolean-operators